### ISSUE TITLE
Trouble Exporting Callosum Tracts

### ISSUE THREAD

**Issue by jdias001** [yeatmanlab/pyAFQ#738]
Following exporting of the AFQ and reco80 tract data, I wanted to export the data for the callosum. However, I keep running into a KeyError that I do not understand. I can see that the AFQ segmentation algorithm has already segmented and cleaned that callosum tracts. Now I simply want the images of the tracts and the tract profiles.

See attachment for my code.

Thank you.
[pyAFQ_TEST_4.pdf](https://github.com/yeatmanlab/pyAFQ/files/7049132/pyAFQ_TEST_4.pdf)

**Comment by 36000** [comment]
Hi! Sorry for the late response. Did you previously run the standard AFQ in the same directory you are now using the the callosum bundles? If so, pyAFQ is trying to use the segmentation files from the standard pyAFQ run (which do not have the callosum bundles) and is running into that error.

**Comment by jdias001** [comment]
I created a new directory to process the colossal tracts and ran into a new error:

`UnboundLocalError                         Traceback (most recent call last)
C:\Users\JAMESW~1.DIA\AppData\Local\Temp/ipykernel_56872/1560466990.py in <module>
     89 #myafq.xforms        = True  #Whether to ouptu the reg_template image in subject space. Default: True.
     90 #myafq.indiv         = True  #Whether to ouput individual bundles in their own files. Default: True.
---> 91 myafq.export_all()
     92 
     93 #Use to rename dwi profiles respective of the segmentation algorithm used.

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\AFQ\api.py in export_all(self)
   2221         self.viz_bundles(export=True)
   2222         if self.seg_algo == "afq":
-> 2223             self.viz_ROIs(export=True)
   2224             self.export_rois()
   2225         all_sub_sess = time() - start_time

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\AFQ\api.py in viz_ROIs(self, bundle_names, export, volume, xform_volume, color_by_volume, cbv_lims, xform_color_by_volume, volume_opacity, n_points, inline, interactive)
   2169                  inline=False,
   2170                  interactive=False):
-> 2171         return self.data_frame.apply(
   2172             self._viz_ROIs,
   2173             axis=1,

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\pandas\core\frame.py in apply(self, func, axis, raw, result_type, args, **kwds)
   7766             kwds=kwds,
   7767         )
-> 7768         return op.get_result()
   7769 
   7770     def applymap(self, func, na_action: Optional[str] = None) -> DataFrame:

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\pandas\core\apply.py in get_result(self)
    183             return self.apply_raw()
    184 
--> 185         return self.apply_standard()
    186 
    187     def apply_empty_result(self):

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\pandas\core\apply.py in apply_standard(self)
    274 
    275     def apply_standard(self):
--> 276         results, res_index = self.apply_series_generator()
    277 
    278         # wrap results

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\pandas\core\apply.py in apply_series_generator(self)
    288             for i, v in enumerate(series_gen):
    289                 # ignore SettingWithCopy here in case the user mutates
--> 290                 results[i] = self.f(v)
    291                 if isinstance(results[i], ABCSeries):
    292                     # If we have a view on v, we need to make a copy because

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\pandas\core\apply.py in f(x)
    108 
    109             def f(x):
--> 110                 return func(x, *args, **kwds)
    111 
    112         else:

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\AFQ\api.py in _viz_ROIs(self, row, bundle_names, export, inline, interactive, volume, xform_volume, color_by_volume, cbv_lims, xform_color_by_volume, volume_opacity, n_points)
   1725                                     targ[:, 2]] = 1
   1726                         warped_roi = auv.transform_inverse_roi(
-> 1727                             aal_roi,
   1728                             self._mapping(row),
   1729                             bundle_name=bundle_name)

UnboundLocalError: local variable 'aal_roi' referenced before assignment`

**Comment by arokem** [comment]
What version of pyAFQ are you using? What do you see when you execute the following?

```
python -c"import AFQ; print(AFQ.__version__)"
```


I think that issue has been resolved in more recent versions.

**Comment by jdias001** [comment]
The version of pyAFQ on my system is 0.8.

**Comment by arokem** [comment]
Are you sure that's the version you are also running in that environment? The reason I ask is that the error seems to be emitted in line 1727 of the api.py module (as the traceback suggests), but the api.py module that is included in version 0.8 only has 1369 lines of code. This looks like the code in that line in version 0.7 

Can you try upgrading the AFQ installation within that environment? For example, if you are running in Jupyter, try running a cell with: 

```
!pip install -U pyAFQ
```

**Comment by jdias001** [comment]
I upgraded my version of pyAFQ, but now I get a completely different error. Mind I am install and running all packages with administrative privileges.

WARNING:py.warnings:The ability to pass arguments to BIDSLayout that control indexing is likely to be removed in future; possibly as early as PyBIDS 0.14. This includes the `config_filename`, `ignore`, `force_index`, and `index_metadata` arguments. The recommended usage pattern is to initialize a new BIDSLayoutIndexer with these arguments, and pass it to the BIDSLayout via the `indexer` argument.
c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\outdated\utils.py:14: OutdatedPackageWarning: The package pingouin is out of date. Your version is 0.3.12, the latest is 0.4.0.
Set the environment variable OUTDATED_IGNORE=1 to disable these warnings.
  return warn(

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\git\__init__.py in <module>
     86 try:
---> 87     refresh()
     88 except Exception as exc:

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\git\__init__.py in refresh(path)
     75 
---> 76     if not Git.refresh(path=path):
     77         return

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\git\cmd.py in refresh(cls, path)
    305                     else:
--> 306                         raise ImportError(err)
    307                 else:

ImportError: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet


The above exception was the direct cause of the following exception:

ImportError                               Traceback (most recent call last)
<ipython-input-1-66e8794a551e> in <module>
      1 #FOR RECO80 SEGMENTATION
      2 
----> 3 from AFQ.data import fetch_hcp
      4 import AFQ.api as api
      5 import AFQ.definitions.mask as afm

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\AFQ\__init__.py in <module>
      2 from .data import *  # noqa
      3 from .utils import *  # noqa
----> 4 from .api import *  # noqa
      5 
      6 _ga_id = "UA-156363454-3"

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\AFQ\api.py in <module>
     33 from importlib import import_module
     34 from collections.abc import MutableMapping
---> 35 import afqbrowser as afqb
     36 
     37 from bids.layout import BIDSLayout

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\afqbrowser\__init__.py in <module>
      1 from .browser import *  # noqa
----> 2 from .gh_pages import *  # noqa

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\afqbrowser\gh_pages.py in <module>
      6 
      7 import github as gh
----> 8 import git
      9 
     10 

c:\users\james w. dias\appdata\local\programs\python\python39\lib\site-packages\git\__init__.py in <module>
     87     refresh()
     88 except Exception as exc:
---> 89     raise ImportError('Failed to initialize: {0}'.format(exc)) from exc
     90 #################

ImportError: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet

**Comment by arokem** [comment]
Oh - that's not good. I think that when we integrated AFQ-Browser (in https://github.com/yeatmanlab/pyAFQ/pull/703) we also made git a dependency for pyAFQ. Maybe we should back off that and make the AFQ-Browser integration optional?

**Comment by arokem** [comment]
For the time being, @jdias001 : do you have git installed? Does calling git on the command line work without error?

**Comment by jdias001** [comment]
I installed git and things seem to be working well currently.

**Comment by arokem** [comment]
Thanks for all the feedback! Let's leave this open for now, until we resolve the git dependency.

**Comment by arokem** [comment]
I believe this was resolved in #740
